core/src/main/java/org/apache/iceberg/BaseReplacePartitions.java

```diff
@@ -53,6 +61,45 @@ public ReplacePartitions validateAppendOnly() {
     return this;
   }
 
+  @Override
+  public ReplacePartitions validateFromSnapshot(long snapshotId) {
+    this.startingSnapshotId = snapshotId;
+    return this;
+  }
+
+  @Override
+  public ReplacePartitions validateNoConflictingAppends() {
+    this.validateNoConflictingAppends = true;
+    return this;
+  }
+
+  @Override
+  public void validate(TableMetadata currentMetadata) {
+    if (validateNoConflictingAppends) {
+      Expression conflictDetectionFilter;
+      if (writeSpec().fields().size() <= 0) {
```

Comment:
You should be able to call `isUnpartitioned()`

Reply:
Done